webapp/tests/karma/ts/modules/reports/reports-content.component.spec.ts

```diff
@@ -490,5 +498,25 @@ describe('Reports Content Component', () => {
       expect(component.selectMode).to.be.false;
       expect(searchFiltersService.freetextSearch.calledOnce).to.be.true;
     }));
+
+    it('should collect telemetry for the selected search results', fakeAsync(() => {
+      store.overrideSelector(Selectors.getSelectMode, true);
+      store.refreshState();
+      fixture.detectChanges();
+
+      expect(searchTelemetryService.recordReportSearch.callCount).to.equal(0);
+      const search = 'case_id:abc-1234';
+      store.overrideSelector(Selectors.getFilters, { search });
+      store.refreshState();
+      fixture.detectChanges();
+      expect(searchTelemetryService.recordReportSearch.callCount).to.equal(0);
+
+      const report = { _id: 'report_id', doc: { case_id: 'abc-1234' } };
+      store.overrideSelector(Selectors.getSelectedReport, report);
+      store.refreshState();
+      fixture.detectChanges();
+      expect(searchTelemetryService.recordReportSearch.callCount).to.equal(1);
+      expect(searchTelemetryService.recordReportSearch.getCall(0).args).to.deep.equal([report.doc, search]);
+    }));
```

Comment:
Is it possible to also cover the edge cases here? 

- Multiple reports selected
- Same report re-selected